src/hooks/useLoadingApi.ts

```diff
@@ -29,10 +35,18 @@ export default function useLoadingApi(): boolean {
   } = useSourceTarget();
 
   const [areReady, setAreReady] = useState(false);
+  const [isSourceReady, setIsSourceReady] = useState(false);
```

Comment:
Can you please elaborate why are we replicating the value coming from `useSourceTarget` to this local state?

Reply:
Makes importing and re-usability easy. Imagine you have a file that both areReady and each one API state state is needed.
You will have to import both `useLoading` and `useSourceTarget` hooks in order to get all Values.
As soon as useLoading hook exist and is used in order to ensure that API are ready for usage - then it made much more sense to extend it and reuse it than import yet another hook to retrieve info that I could get from the same hook

Reply:
The 2 extra `useState`s for that are not necessary, we can simply return `isSourceApiReady` and `isTargetApiReady` (we do need the state variable for `areReady` though).

Reply:
I prefer to know the init state upon first call of the hook (false) instead of initializing the states from another state that is not known at the moment. The performance is the same as the 1st useEffect will run anyway and more control is provided.
If we do not want that then no state is needed for any of the cases. We can just: 
```
return {
    areApiReady: isSourceApiReady && isTargetApiReady,
    sourceReady: isSourceApiReady,
    targetReady: isTargetApiReady
  };
  ```
  but I do not like it as it is too float-y

Reply:
> The 2 extra `useState`s for that are not necessary, we can simply return `isSourceApiReady` and `isTargetApiReady` (we do need the state variable for `areReady` though).

This is my exactly my point.

Reply:
> but I do not like it as it is too float-y. 

Can you elaborate what problem exactly this creates? I'm abstracting from all the bloat that comes from (`useSourceTarget -> Context -> `connections` passed as props -> `useConnections` hook), but essentially we already have a "state" that we maintain in the top level component that knows if api is ready or not. The hook that you propose is creating local copies of these variables, so now instead of having one "source of truth" we have two and you have to use `useEffect` to keep them in sync.
Obviously, we can argue that the thanks to this the state is more local and it's maybe easier to reason about the hook itself, but long term it's much harder to reason about the entire application - this is just covering bloated code with even more bloated code and we will easily loose track of what is happening where. These kind of things lead to subtle bugs (one component thinks "ready", while other is "not ready") and I feel that historically a solution to these kinds of bugs is just adding more bloat (we miss `useEffect` here, or a dependency in existing `useEffect`, etc).

Reply:
Agreed and understood your point. I will remove the non-needed useState/useEffect from this component. 
At the end - this component is just a way to "receive" simpler and faster the states of API across the app.